File: specs-capstone-backend/src/main/java/spec/alissa/backend/models/AdoptionForm.java
package spec.alissa.backend.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdoptionForm {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long formId;
    @Column(nullable = false)
    private String animalName;
    @Column(nullable = false)
    private String animalBreed;
    @Column(nullable = false)
    private String fullName;
    @Column(nullable = false)
    private String phoneNumber;
    @Column(nullable = false)
    private String email;
    private String birthdate;
    @Column(nullable = false)
    private String occupation;
    @Column(nullable = false)
    private String streetAddress;
    @Column(nullable = false)
    private String city;
    @Column(nullable = false)
    private String state;
    @Column(nullable = false)
    private String zipCode;
    @Column(nullable = false)
    private String country;
    @Column(nullable = false)
    private boolean hasPet;
}

File: specs-capstone-frontend/public/adoption-form.js
const form = document.querySelector("adoption-form-flex");
const button = document.getElementById("adoption-submit");

function buttonClicked() {
    const animalName = document.querySelector("#adp-name");
    const animalBreed = document.querySelector("#adp-species");
    const fullName = document.querySelector("#adp-full-name");
    const phoneNumber = document.querySelector("#adp-number");
    const email = document.querySelector("#adp-email");
    const birthdate = document.querySelector("#adp-date");
    const occupation = document.querySelector("#adp-job");
    const streetAddress = document.querySelector("#adp-address");
    const city = document.querySelector("#city");
    const state = document.querySelector("#state");
    const zipCode = document.querySelector("#zip-code")
    const country = document.querySelector("#country");
    const hasPet = document.querySelector("#pet-boolean");
    let body = {
        "animalName": animalName.value,
        "animalBreed": animalBreed.value,
        "fullName": fullName.value,
        "phoneNumber": phoneNumber.value,
        "email": email.value,
        "birthdate": birthdate.value,
        "occupation": occupation.value,
        "streetAddress": streetAddress.value,
        "city": city.value,
        "state": state.value,
        "zipCode": zipCode.value,
        "country": country.value,
        "hasPet": hasPet.value === "yes"
    }
    adoptionRequest(body);
    location.replace("index.html");
    alert("You submitted an adoption form!");
}

function adoptionRequest(body) {
    axios.post(`${baseURL}/api/adoption-form/add/${user.userId}`, body).then(({ data }) => { 
    })
}

button.addEventListener("click", buttonClicked);
